### PR TITLE
nixos/{modules,tests}/teeworlds: init

### PR DESCRIPTION
**Pull request by hax404**
###### Motivation for this change
add module and test for Teeworlds based on the OpenArena module and test

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/games/teeworlds.nix [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        description = ''
+          Password to connect to the server.
+        '';
+      };
+
+      rconPassword = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Password to access the remote console. If not set, a random generated is displayed in the server log.
```
Comment (mweinelt) [inline nixos/modules/services/games/teeworlds.nix:68]:
```suggestion
          Password to access the remote console. If not set, a randomly generated one is displayed in the server log.
```

**Review by mweinelt — APPROVED** [review]

**Comment by hax404** [comment]
resolved race condition in the test and rebased on master

**Comment by Mic92** [comment]
Will have a look.

File: nixos/tests/teeworlds.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      )
+
+      client2.execute("teeworlds 'player_name Bob;connect server'&")
+      server.wait_until_succeeds(
+          'journalctl -u teeworlds -e | grep --extended-regexp -q "team_join player=\'[0-9]:Bob"'
+      )
+
+      server.sleep(10)  # wait for a while to get a nice screenshot
+
+      client1.screenshot("screen_client1")
```
Comment (Mic92) [inline nixos/tests/teeworlds.nix:51]: I like this part.

**Pull request merged**